src/traces/streamtube/calc.js

```diff
@@ -206,6 +206,9 @@ function distinctVals(col) {
 function filter(arr, len) {
     if(len === undefined) len = arr.length;
 
+    // no need for casting typed arrays to numbers
+    if(Lib.isTypedArray(arr)) return arr.slice(0, len);
```

Comment:
It's better to use `arr.subarray()` for typed arrays, as `.slice` isn't supported in older browsers.

Reply:
Good call.
Wondering if we need to slice the array at all?

Reply:
`slice` (or subarry) is needed for `distictVals` to work correctly.
But it may be better not to slice if the length is correct.

Reply:
> But it may be better not to slice if the length is correct.

that could be a nice optimisation for typed arrays, but for regular arrays we'll need to slice in general to cast numerical-strings items w/o mutating.